### PR TITLE
Add the Str::between function.

### PR DESCRIPTION
**Pull request by mathieutu**
Hey guys!
I made this helper for me, and I think it can help many people.

It's fully tested, ant I think it stays in the _spirit_ of the others functions, and can be added to the core.

What do you think ?

Thx,
Matt'

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -455,6 +455,29 @@ public static function substr($string, $start, $length = null)
     }
 
     /**
+     * Returns the portion of string between the first start and the last end strings.
+     *
+     * @param string $value
+     * @param string $start
```
Comment (m1guelpf) [inline src/Illuminate/Support/Str.php]: Maybe you should mention the default value is null

File: src/Illuminate/Support/Str.php [hunk 2]
```diff
@@ -455,6 +455,29 @@ public static function substr($string, $start, $length = null)
...
+     * @param string $end
```
Comment (m1guelpf) [inline src/Illuminate/Support/Str.php]: Maybe you should mention the default value is null

File: src/Illuminate/Support/Str.php (hunk 1 above)
Comment (mathieutu) [inline src/Illuminate/Support/Str.php]: True! It's done, thx.

File: src/Illuminate/Support/Str.php (hunk 2 above)
Comment (mathieutu) [inline src/Illuminate/Support/Str.php]: True! It's done, thx.

File: tests/Support/SupportStrTest.php [hunk 3]
```diff
@@ -234,6 +234,20 @@ public function testSubstr()
...
+    public function testBetween()
+    {
+        $this->assertEquals('bar', Str::between('foobarbaz', 'foo', 'baz'));
+        $this->assertEquals('foobar', Str::between('foobarbaz', null, 'baz'));
+        $this->assertEquals('foobar', Str::between('foobarbaz', '', 'baz'));
+        $this->assertEquals('barbaz', Str::between('foobarbaz', 'foo', null));
+        $this->assertEquals('barbaz', Str::between('foobarbaz', 'foo', ''));
+        $this->assertEquals('', Str::between('foobarbaz', 'foo', 'nope'));
+        $this->assertEquals('', Str::between('foobarbaz', 'nope', 'baz'));
+        $this->assertEquals('', Str::between('foobarbaz', 'nope', 'baz'));
```
Comment (fernandobandeira) [inline tests/Support/SupportStrTest.php]: Theres 2 equal tests here, 245 and 246...

**Comment by taylorotwell** [comment]
I don't typically add a lot of helper functions that the framework doesn't use.

**Comment by mathieutu** [comment]
OK I'd have tried.
Maybe a day it will need it..? :-)

**Pull request closed without merging**